### ISSUE TITLE
[CLOSED] Boost accumulator median needs 5 data points

### ISSUE THREAD

**Issue by samhatchett** [OpenWaterAnalytics/epanet-rtx#59]
<a href="https://github.com/jamesuber"><img src="https://avatars.githubusercontent.com/u/2408457?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [jamesuber](https://github.com/jamesuber)**
_Monday Oct 06, 2014 at 00:02 GMT_
_Originally opened as https://github.com/CitiLogics/epanet-rtx/issues/18_

---

Fascinating that even the computation of the median of a series, when approached at an expert level and an eye to efficiency, is not straightforward: https://svn.boost.org/trac/boost/ticket/6992

I think this explains some things I've been seeing using the median filter that just didn't look right. I did confirm that using boost median with three data points always just gives you the last one. 

Bottom line is that we shouldn't use the median unless we think there will be at least 5 points.

**Comment by samhatchett** [comment]
<a href="https://github.com/samhatchett"><img src="https://avatars.githubusercontent.com/u/2386549?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [samhatchett](https://github.com/samhatchett)**
_Monday Oct 06, 2014 at 00:05 GMT_

---

Right. It's a median "estimate"

**Comment by samhatchett** [comment]
<a href="https://github.com/jamesuber"><img src="https://avatars.githubusercontent.com/u/2408457?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jamesuber](https://github.com/jamesuber)**
_Monday Oct 06, 2014 at 00:16 GMT_

---

stating the obvious can be dangerous ;-)
 JIM UBER • 513.368.6303 • CITILOGICS.COM

**Comment by samhatchett** [comment]
<a href="https://github.com/samhatchett"><img src="https://avatars.githubusercontent.com/u/2386549?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [samhatchett](https://github.com/samhatchett)**
_Tuesday Oct 07, 2014 at 20:08 GMT_

---

yep, boost is not exact. 5 points is good, less is bad, more is an estimate.
